app/controllers/recipe_foods_controller.rb

```diff
@@ -0,0 +1,40 @@
+class RecipeFoodsController < ApplicationController
+  def new
+    @recipe = Recipe.find(params[:recipe_id])
+    @foods = current_user.foods
+    @recipe_food = RecipeFood.new
+  end
+
+  def create
+    @recipe = Recipe.find_by(id: params[:recipe_id])
+    food_ids = RecipeFood.where(recipe_id: @recipe.id).map(&:food_id)
+
+    if food_ids.include?(recipe_food_params[:food_id].to_i)
+      return redirect_to recipe_path(id: @recipe.id),
+                         alert: 'Recipe already has this ingredient. Use Modify button'
+    end
+
+    @recipe_food = @recipe.recipe_foods.build(recipe_food_params)
+
+    if @recipe_food.save
+      redirect_to recipe_path(id: @recipe.id), notice: 'A new ingredient has been added successfully'
+    else
+      flash[:alert] = 'Something went wrong, Try again!'
+      render :new
+    end
+  end
+
+  def destroy
+    @recipe_food = RecipeFood.includes(:recipe).find_by(id: params[:id])
+
+    if @recipe_food.destroy
+      redirect_to recipe_path(id: params[:recipe_id]), notice: 'The ingedrient was successfully removed.'
+    else
+      redirect_to recipe_path(id: params[:recipe_id]), notice: 'Error in removing the ingredient.'
+    end
+  end
+
+  def recipe_food_params
+    params.require(:recipe_food).permit(:quantity, :food_id)
+  end
```

Comment:
- [ ] @Team great work adding a method to handle parameters, however, it's recommended to move the methods to be private, that way your app will be more secure.

Reply:
### **Thank you for your valuable feedback😍!**
We appreciate your suggestion to make the method private for better security. We will definitely take this into account and update the code accordingly in the next commit. Thank you again for your helpful review and for helping us improve our app!